python/cudf/cudf/tests/test_setitem.py

```diff
@@ -45,6 +45,23 @@ def test_dataframe_setitem_existing_columns(df, arg, value):
     assert_eq(df, gdf, check_dtype=False)
 
 
+@pytest.mark.parametrize(
+    ("df", "arg", "value", "expected"),
+    [
+        (DataFrame(), ["a", "b"], 0, DataFrame({"a": [], "b": []})),
+        (
+            DataFrame({"a": [1, 2, 3]}),
+            ["b"],
+            9,
+            DataFrame({"a": [1, 2, 3], "b": [9, 9, 9]}),
+        ),
+    ],
+)
+def test_dataframe_setitem_non_existing_columns(df, arg, value, expected):
```

Comment:
Shouldn't we be testing against Pandas here?

Reply:
pandas throws error in this case, but this was supported by old cudf before I made changes `__settitem__`.